javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistClassDeclaration.java

```diff
@@ -309,17 +324,23 @@ public boolean isClass() {
     }
 
     @Override
-    public ResolvedReferenceType getSuperClass() {
+    public Optional<ResolvedReferenceType> getSuperClass() {
         try {
-            if (ctClass.getClassFile().getSuperclass() == null) {
-                return new ReferenceTypeImpl(typeSolver.solveType(Object.class.getCanonicalName()), typeSolver);
+            if ("java.lang.Object".equals(ctClass.getClassFile().getName())) {
```

Comment:
hmm... I remember there being a reason that I left this string comparison in place...

perhaps it can be gotten around by doing something like `javaLangObject().getQualifiedName()` or similar... 

leaving this note here as a reminder to investigate :)

Reply:
personal (very weakly held) conclusion: leave as-is, and if anyone else wants to edit it then this is welcomed